pkg/kubelet/dockershim/security_context.go

```diff
@@ -164,9 +166,25 @@ func modifyHostNetworkOptionForContainer(hostNetwork bool, sandboxID string, hc
 	hc.NetworkMode = dockercontainer.NetworkMode(sandboxNSMode)
 	hc.IpcMode = dockercontainer.IpcMode(sandboxNSMode)
 	hc.UTSMode = ""
-	hc.PidMode = ""
+	hc.PidMode = dockercontainer.PidMode(sandboxNSMode)
 
 	if hostNetwork {
 		hc.UTSMode = namespaceModeHost
 	}
 }
+
+// modifyPIDNamespaceOverrides implements two overrides for the default PID namespace sharing in Docker:
+//     1. Docker engine prior to API Version 1.24 doesn't support attaching to another container's
+//        PID namespace. This check can be removed when Kubernetes' minimum Docker version advances
+//        past 1.11 (API version 1.23).
+//     2. The administrator has overridden the default behavior by means of the --disable-docker-shared-pid
+//        kubelet flag. This is an "escape hatch" to return to previous behavior of isolated namespaces
+//        and should be removed once no longer needed.
+func modifyPIDNamespaceOverrides(disableSharedPID bool, version *semver.Version, hc *dockercontainer.HostConfig) {
+	if !strings.HasPrefix(string(hc.PidMode), "container:") {
+		return
+	}
+	if disableSharedPID || (version.Major <= 1 && version.Minor <= 23) {
```

Comment:
Use the existing helper function for comparing the versions: https://github.com/kubernetes/kubernetes/blob/v1.6.0-beta.0/pkg/kubelet/dockershim/helpers.go#L314